services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/FileScanService.kt

```diff
@@ -0,0 +1,86 @@
+package fi.hel.haitaton.hanke.security
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import fi.hel.haitaton.hanke.toJsonString
+import java.time.Duration.ofSeconds
+import mu.KotlinLogging
+import org.apache.commons.lang3.BooleanUtils.isNotFalse
+import org.springframework.core.io.ByteArrayResource
+import org.springframework.http.MediaType.APPLICATION_JSON
+import org.springframework.http.MediaType.APPLICATION_PDF
+import org.springframework.http.MediaType.MULTIPART_FORM_DATA
+import org.springframework.http.client.MultipartBodyBuilder
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.BodyInserters.fromMultipartData
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+
+private val logger = KotlinLogging.logger {}
+
+private const val FORM_KEY = "FILES"
+
+@Service
+class FileScanService(
+    private val fileScanClient: WebClient,
+) {
+
+    fun scanFiles(files: Set<Pair<String, ByteArray>>): FileScanResponse {
+        logger.info { "Scanning ${files.size} files." }
+        val result = getResults(files).also { response -> response.validateStatus() }
+        return result.also { logStatus(it) }
+    }
+
+    private fun getResults(files: Set<Pair<String, ByteArray>>): FileScanResponse {
+        val data =
+            MultipartBodyBuilder()
+                .apply {
+                    files.forEach { (name, bytes) ->
+                        part(FORM_KEY, ByteArrayResource(bytes), APPLICATION_PDF).filename(name)
+                    }
+                }
+                .build()
+
+        return fileScanClient
+            .post()
+            .uri("/api/v1/scan")
+            .contentType(MULTIPART_FORM_DATA)
+            .accept(APPLICATION_JSON)
+            .body(fromMultipartData(data))
+            .retrieve()
+            .bodyToMono(FileScanResponse::class.java)
+            .timeout(ofSeconds(60))
+            .doOnError(WebClientResponseException::class.java) {
+                logger.error { "Error uploading file: $it" }
+            }
+            .blockOptional()
+            .orElseThrow()
+    }
+
+    private fun FileScanResponse.validateStatus() {
+        if (!success) {
+            throw FileScanException("Scan failed, result: ${this.toJsonString()}")
+        }
+    }
+
+    private fun logStatus(result: FileScanResponse) {
+        if (result.hasInfected()) {
+            logger.error { "Infected file detected, scan result: ${this.toJsonString()}" }
```

Comment:
Is this really an error? This could be a warning since we detect and handle the exceptional flow appropriately.

Dumping the whole result as JSON might not be the most useful information for someone reading the logs.

Reply:
```
    private fun logStatus(results: List<FileResult>) {
        if (results.hasInfected()) {
            results.filterInfected().forEach {
                logger.warn { "Infected file detected, scan result: ${it.toJsonString()}" }
            }
        } else {
            logger.info { "Files scanned successfully." }
        }
    }
    
```
Changed approach to filter infected file results and log them separately. It is still in json format, but I think it is readable enough.